### ISSUE TITLE
Disable dangerous checkpoint deletion behaviour

### ISSUE THREAD

**Issue by thefiddler** [Lightning-AI/lightning#17189]
### Description & Motivation

When PL resumes training from a checkpoint, it deletes the old checkpoint from the version directory. Over the last several weeks, this behaviour has led to loss of several days of training:
- resume old run, NaN loss occurs in new run -> new run checkpoint unusable, but old checkpoint has been removed -> have to start training from scratch.
- new run contains problematic samples -> new run checkpoint unusable, but old checkpoint removed -> have to start training from scratch.
- out of disk space -> new run checkpoint fails to write, but old checkpoint already deleted -> have to start training from scratch.
- pretrained model -> want to finetune several variations, but PL keeps deleting the pretrained checkpoint -> have to work around PL to achieve this.

### Pitch

Is there a way to disable this behaviour or at least make it opt-in? (I did not find it in the documentation.)

The value of a checkpoint that took weeks to train is MUCH higher than saving a few hundred MB of disk space by deleting resumed checkpoints.

### Alternatives

I have resorted to making copies of checkpoints manually via the filesystem, but this is error prone and time-consuming.

cc @borda @tchaton @justusschock @awaelchli @carmocca

**Comment by carmocca** [comment]
I think it makes sense to avoid deleting the checkpoint originally passed to `trainer.fit(..., ckpt_path="this/checkpoint/should/not/be/deleted")`
